sdk/inc/azure/core/internal/az_event_pipeline_internal.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file
+ *
+ * @brief Definition of #_az_event_pipeline and related types describing a bi-directional HFSM
+ *        pipeline.
+ *
+ */
+
+#ifndef _az_EVENT_PIPELINE_INTERNAL_H
+#define _az_EVENT_PIPELINE_INTERNAL_H
+
+#include <azure/core/az_event.h>
+#include <azure/core/az_event_policy.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <stdint.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+// Definition is below.
+
+typedef struct _az_event_pipeline _az_event_pipeline;
```

Comment:
I don't think this is necessary (anymore). No self-reference here: we can combine the `typedef` with the struct declaration below?